lib/fog/ovirt/requests/compute/v4/add_volume.rb

```diff
@@ -0,0 +1,53 @@
+module Fog
+  module Compute
+    class Ovirt
+      class V4
+        class Real
+          DISK_SIZE_TO_GB = 1024 * 1024 * 1024
+
+          def add_volume(id, options = {})
+            raise ArgumentError, "instance id is a required parameter" unless id
+
+            options = add_options_defaults(options)
+
+            disk_attachments_service = client.system_service.vms_service.vm_service(id).disk_attachments_service
+            disk = OvirtSDK4::DiskAttachment.new(options)
+            disk_attachments_service.add(disk)
+          end
+
+          # rubocop:disable Metrics/AbcSize
+          def add_options_defaults(options)
+            search = options[:search] || format("datacenter=%<datacenter>s", :datacenter => datacenter)
```

Comment:
Since we are modifying the options argument, we should do `options = options.dup` at the beginning of the method